File: javascript/diagonal_negativos.js
const n = parseInt(prompt("Qual a ordem da matriz?"));

let matriz = [];

let qtdNegativos = 0;

for(i = 0; i < n; i++) {
    matriz.push([]);
}

for(i = 0; i < n; i++) {
    for(j = 0; j < n; j++) {
        matriz[i][j] = parseInt(prompt("Elemento [" + i + "," + j + "]:"));

        if (matriz[i][j] < 0) {
            qtdNegativos++;
        }
    }
}

console.log("DIAGONAL PRINCIPAL:");

for(i = 0; i < n; i++) {
    for(j = 0; j < n; j++) {
        if (i == j) {
            console.log(matriz[i][j]);
        }
    }
}

console.log("QUANTIDADE DE NEGATIVOS = %d", qtdNegativos);

File: python/diagonal_negativos.py
def criar_matriz(num_linhas: int, num_colunas: int) -> list[list[int]]:
    m = []

    for i in range(num_linhas):
        m.append([0] * num_colunas)

    return m


def ler_matriz(matriz: list[list[int]]) -> None:
    for i in range(len(matriz)):
        for j in range(len(matriz[0])):
            matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))


n = int(input("Qual a ordem da matriz? "))
matriz = criar_matriz(n, n)
ler_matriz(matriz)

print("DIAGONAL PRINCIPAL:")
for i in range(n):
    print(matriz[i][i], end=" ")

qtd_negativos = 0
for i in range(n):
    for j in range(n):
        if matriz[i][j]:
            qtd_negativos += 1

print("QUANTIDADE DE NEGATIVOS =", qtd_negativos)
